# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/cdrtools/chrp-boot.patch
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -ur cdrtools-2.00.3/mkisofs/mkisofs.c cdrtools-2.00.3-local/mkisofs/mkisofs.c
--- cdrtools-2.00.3/mkisofs/mkisofs.c	2003-05-15 15:12:44.000000000 +0100
+++ cdrtools-2.00.3-local/mkisofs/mkisofs.c	2003-07-30 21:24:16.000000000 +0100
@@ -401,6 +401,8 @@
 #define OPTION_HFS_BLESS		2040
 #define OPTION_HFS_PARMS		2041
 
+#define OPTION_CHRP_BOOT		2042
+
 #endif	/* APPLE_HYB */
 
 static int	save_pname = 0;
@@ -2096,6 +2100,21 @@
 				exit(1);
 			}
 			break;
+		case OPTION_CHRP_BOOT:
+			if (use_chrp_boot)
+				break; /* silently allow duplicates */
+			use_chrp_boot = 1;
+			if (use_prep_boot > 3) {
+#ifdef USE_LIBSCHILY
+				comerrno(EX_BAD,
+				"Maximum of 4 partition entries are allowed\n");
+#else
+				fprintf(stderr,
+				"Maximum of 4 partition entries are allowed\n");
+#endif
+				exit(1);
+			}
+			break;
 #endif	/* PREP_BOOT */
 #endif	/* APPLE_HYB */
 		default:
